### PR TITLE
(DEV-564) computeShapeInfo fixes, Extent calculation optimization

### PR DESCRIPTION
**Pull request by sabrina-shanman**
https://highfidelity.atlassian.net/browse/DEV-564

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Can someone from High Fidelity approve this PR?

File: libraries/entities-renderer/src/RenderableModelEntityItem.cpp [hunk 1]
```diff
@@ -380,59 +380,43 @@ void RenderableModelEntityItem::computeShapeInfo(ShapeInfo& shapeInfo) {
...
+        size_t numParts = 0;
+        for (const HFMMesh& mesh : collisionGeometry.meshes) {
+            numParts += mesh.triangleListMesh.parts.size();
+        }
+        pointCollection.reserve(numParts);
 
         // the way OBJ files get read, each section under a "g" line is its own meshPart.  We only expect
         // to find one actual "mesh" (with one or more meshParts in it), but we loop over the meshes, just in case.
-        foreach (const HFMMesh& mesh, collisionGeometry.meshes) {
+        for (const HFMMesh& mesh : collisionGeometry.meshes) {
```
Comment (samcake) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: WE re still itereating over  the meshes, not the shapes ?

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22353/HighFidelity-Beta-PR16460-33bb0e2.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22353/HighFidelity-Quest-Beta-PR16460-33bb0e2.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22353/HighFidelity-Beta-PR16460-33bb0e2.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22353/HighFidelity-Beta-Interface-PR16460-33bb0e2.exe

**Comment by sabrina-shanman** [comment]
This PR needs more work.

**Comment by sabrina-shanman** [comment]
In the interest of time, let's merge this somewhat less ideal implementation. See branch https://github.com/sabrina-shanman/hifi/tree/instancing_collision_new-and-improved

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22399/HighFidelity-Beta-PR16460-33bb0e2.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22399/HighFidelity-Quest-Beta-PR16460-33bb0e2.apk)

**Pull request merged**